src/shogun/regression/GLM.h

```diff
@@ -0,0 +1,112 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Ahmed Khalifa
+ */
+/*
+References:
+https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/family
+https://www.statsmodels.org/stable/generated/statsmodels.genmod.generalized_linear_model.GLM.html#statsmodels.genmod.generalized_linear_model.GLM
+http://glm-tools.github.io/pyglmnet/api.html
+*/
+#ifndef _GENERALIZEDLINEARMODEL_H__
+#define _GENERALIZEDLINEARMODEL_H__
+
+#include <shogun/lib/config.h>
+
+#include <shogun/features/Features.h>
+#include <shogun/machine/FeatureDispatchCRTP.h>
```

Comment:
good point, would be good if this class worked for multiple feature types. But you are not using this atm I think?

Reply:
I haven't explored all the feature types aren't all features the same? (aka. just numbers at the end of the day) please do enlighten me :D

Reply:
read the code :)